lib/foreman_rh_cloud/engine.rb

```diff
@@ -122,7 +122,8 @@ def self.register_scheduled_task(task_class, cronline)
   end
 
   def self.with_iop_smart_proxy?
-    SmartProxy.with_features('iop').exists?
+    return @is_iop unless @is_iop.nil?
+    @is_iop = SmartProxy.with_features('iop').exists?
```

Comment:
This can be one line:
```
def self.with_iop_smart_proxy?
  @is_iop ||= SmartProxy.with_features('iop').exists?
end
```

Reply:
There is a small difference in case is_iop is false: it will retry the smart proxy check every time.

Reply:
`??` solves this in JavaScript. I'm sure it's coming in Ruby soon :)